src/api/instagram.ts

```diff
@@ -254,8 +274,7 @@ export class Instagram {
           window["_sharedData"].entry_data.PostPage[0].graphql,
         );
       });
-      await this.addToPostBuffer(JSON.parse(data));
-
+      parsed = JSON.parse(data) as PostType;
       await postPage.close();
```

Comment:
The error catching I wrote here is probably too general to really be acceptable. However, if kept in its current form, postPage still isn't guaranteed to be closed because the error could occur when parsing the data, and then the postPage will not close and its reference will be lost.

I think of pages in puppeteer like malloc/free in C. You've got to keep track of them and not let them pile up.

Reply:
Fixed with an additional try/catch block

Reply:
In commit `12b4f1d8524a1765035a62ef1c31b8777e857b85` I've reverted back the change I've made in commit `fc74a790ba85abec4380509392bab3b0f19d05ab`.

I've double checked the code, which was before the commit:

If the error occured in parsing the data, code execution goes to catch block, where `postPage.close()` is.
If not - it will be closed (the code is on the next line).

If data validation will fail, the page is already closed. 

P.S. I've added two tests in strict mode section. They check, that error object message should contain validation information when validation fails. The code in commit `fc74a790ba85abec4380509392bab3b0f19d05ab` was throwing another error (can not close page which is closed), so I've written a regression page for this scenario.

Reply:
You're right. I think I was being tripped up by the code folding in Github's web ui.